src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -176,6 +182,10 @@ public void onTextChanged(CharSequence s, int start, int before, int count) { }
 
             int certCheckFlags = DcHelper.getInt(this, "imap_certificate_checks");
             certCheck.setSelection(certCheckFlags);
+            encryptCheckbox.setHeight(0);
```

Comment:
I'm setting the height to 0 here instead of setting the visibility to GONE because setting the visibility to GONE doesn't work well with constraint layouts: The `inbox` TextView is constrained to `Top_toBottomOf="@id/encrypt_checkbox"` and if the visibility was GONE, this constraint would probably fail, even though I didn't try it out (it did last time I tried to do this with constraintLayout).

(Also, the visibility is changed everytime the user hits "Advanced", so the visibility would have to be changed in the onClickListener, not here.)

Reply:
+1 for an easy solution

ftr, there's a Barrier class that should help in these cases: https://developer.android.com/reference/androidx/constraintlayout/widget/Barrier